chain/syncer.go

```diff
@@ -111,6 +116,21 @@ func NewSyncer(e syncStateEvaluator, s syncerChainReaderWriter, m MessageProvide
 		stateEvaluator:  e,
 		chainStore:      s,
 		messageProvider: m,
+		clock:           c,
+		status: &SyncerStatus{
+			ValidatedHead:       types.UndefTipSet.Key(),
+			ValidatedHeadHeight: 0,
```

Comment:
I think you can read the above values from the store right away.

Reply:
Not quite, the `chainStore` doesn't know what its head is when we initialize the syncer.

We call NewSyncer in the nodes `Build()` method:
https://github.com/filecoin-project/go-filecoin/blob/83becde01402803454b43ea75c77364d13e28564/node/node.go#L458
When it asks the `chainStore` for its head via `GetHead` it returns an empty tipset because we don't set the `chainStore` head until `Load` is called in node `Start()`:
https://github.com/filecoin-project/go-filecoin/blob/83becde01402803454b43ea75c77364d13e28564/node/node.go#L536

I could plug the genesisCID and height in here since the chainStore can tell us that at node build time, but that will be outdated once the chainStore loads its state.

Reply:
I would prefer to leave the values undefined and wait for them to be updated once the syncer encounters a new tipset, what do you think?

Reply:
I wonder if since this is a somewhat global decoupled object if it should initialized and owned by the syncer? Perhaps it gets initialized on its own (looks like based on the values it wouldn't need any dependencies to write a constructor), and then the ChainReader gets a reference too that it can modify when Load gets called?

Reply:
That sounds like a good idea, will push something up shortly.

Reply:
pushed in https://github.com/filecoin-project/go-filecoin/pull/3305/commits/73b03f6b95ed6c1ccefcf996c9e9e77209d854bf